### DISCUSSION TITLE
How to create a complex polygon

### DISCUSSION THREAD

**Discussion by svaterlaus** [Birch-san/box2d-wasm#29] [Q&A]
I'm trying to create a complex polygon shape with specific vertices but I can't find a method to do so. Other Box2D implementations have methods like `shape.setAsArray`, but I've been trying to figure out how to do the same with this with no luck.

**Comment by svaterlaus** [comment]
Nevermind, I just figured it out using the `tuplesToVec2Array` helper function for the shape.Set() method.

Hopefully this can help someone else in the future.

**Comment by BlueNebulaDev** [comment]
In case someone else struggles with this, here is an example on how to use `tuplesToVec2Array`:

```
const points = [[0,1], ...];
const [vecArr, destroyVecArr] = Box2D.pointsToVec2Array(points);
shape.Set(vecArr, vertices.length);
destroyVecArr();
```

`pointsToVec2Array` is the same, but it expects points represented as `{x,y}` objects:

```
const points = [{x:0, y:1}, ...];
const [vecArr, destroyVecArr] = Box2D.pointsToVec2Array(points);
shape.Set(vecArr, vertices.length);
destroyVecArr();
```
